eZ/Publish/Core/Persistence/Cache/ContentHandler.php

```diff
@@ -343,18 +342,25 @@ public function deleteVersion($contentId, $versionNo)
      */
     public function listVersions($contentId, $status = null, $limit = -1)
     {
-        $cacheItem = $this->cache->getItem("ez-content-${contentId}-version-list" . ($status !== null ? "-byStatus-${status}" : '') . "-limit-{$limit}");
+        // Don't cache non typical lookups to avoid filling up cache and tags with uncommon lookups.
+        if ($status !== null || $limit !== -1) {
+            $this->logger->logCall(__METHOD__, ['content' => $contentId, 'status' => $status]);
+
+            return $this->persistenceHandler->contentHandler()->listVersions($contentId, $status, $limit);
+        }
 
+        // Cache the typical case.
+        $cacheItem = $this->cache->getItem("ez-content-${contentId}-version-list");
         if ($cacheItem->isHit()) {
-            $this->logger->logCacheHit(['content' => $contentId, 'status' => $status]);
+            $this->logger->logCacheHit(['content' => $contentId]);
 
             return $cacheItem->get();
         }
 
-        $this->logger->logCacheMiss(['content' => $contentId, 'status' => $status]);
-        $versions = $this->persistenceHandler->contentHandler()->listVersions($contentId, $status, $limit);
+        $this->logger->logCacheMiss(['content' => $contentId]);
+        $versions = $this->persistenceHandler->contentHandler()->listVersions($contentId);
```

Comment:
```suggestion
        $versions = $this->persistenceHandler->contentHandler()->listVersions($contentId, $status, $limit);
```

_Default values for parameters might be changed by concrete implementation_